### PR TITLE
[Azure Search] Fix the definition of WebApiSkill

### LINKED ISSUE #6468: [Azure Search][BUG] Trying to create a skillset with a WebApiSkill fails with "request is invalid"
**Describe the bug**
Trying to create a skillset with a custom WebApiSkill fails with an error.

***Exception or Stack Trace***
Microsoft.Rest.Azure.CloudException : The request is invalid. Details: skillset : Object reference not set to an instance of an object.

**To Reproduce**
Add a WebApiSkill to a Skillset and attempt to call searchClient.Skillsets.Create(skillset).

**Expected behavior**
The skillset should be successfully created or should fail with a more obvious message of what is wrong.

**Additional context**
In debugging this issue with the SDK myself, I realized that the issue is because the object for WebApiHttpHeaders within the WebApiSkill adds an extra layer of object encapsulation in the httpHeaders field that should not be there.

Currently we send:
`"httpHeaders": {
        "headers": {
                //headers
        }
}`
But we should actually send:
`"httpHeaders": {
       // headers
},`

In addition, should the httpHeaders and httpMethod fields be required?
Also, missing test coverage

### PR DESCRIPTION
**Pull request by arv100kri**
This SDK was generated based on [this completed pull request](https://github.com/Azure/azure-rest-api-specs/pull/6779) in the swagger repository.

- This pull request, intends to correct the definition of the `WebApiSkill` model in the SDK, by making headers a simple dictionary within the skill and not a nested property.

- Added a test to ensure correctness

- This is a breaking change, and is already preceeded by a version bump in [this PR](https://github.com/Azure/azure-sdk-for-net/pull/7031)

Fixes #6468

### CODE REVIEW

**Review by Careyjmac — CHANGES_REQUESTED** [review]

**Review by Careyjmac — APPROVED** [review]

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs [hunk 1]
```diff
@@ -23,6 +24,26 @@ public sealed class SkillsetsTests : SearchTestBase<SearchServiceFixture>
...
+            });
+        }
+
+        [Fact]
+        public void CreateSkillsetReturnsCorrectDefinitionWebApiSkillWithoutHeaders()
+        {
+            Run(() =>
+            {
+                SearchServiceClient searchClient = Data.GetSearchServiceClient();
+                CreateAndValidateSkillset(searchClient, CreateTestSkillsetWebApiSkill(false));
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: nit: Please explicitly name parameters when passing in magic constants.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs [hunk 2]
```diff
@@ -63,18 +84,6 @@ public void CreateSkillsetReturnsCorrectDefinitionOcrHandwritingSentiment()
             });
         }
 
-        [Fact]
-        public void CreateSkillsetThrowsExceptionWithInvalidLanguageSelection()
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: Why is this test being removed?

**Review by bitsforthought — CHANGES_REQUESTED** [review]
I have a concern about an apparently unrelated test change.

File: sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs (hunk 2 above)
Comment (Careyjmac) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: A change in the backend makes this test unnecessary, so I asked Arvind to remove it as part of this change (it currently fails if you run it actually).  We can do it in a separate PR if you really want.

Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: Also, there is a separate task on me to obsolete-ize the `textExtractionAlgorithm` as it's no longer being used (and hence making the test fail as it's not validated in the service)

Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: > A change in the backend makes this test unnecessary, so I asked Arvind to remove it as part of this change (it currently fails if you run it actually). We can do it in a separate PR if you really want.

nit: The test fails if you run it in -Record mode

Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search/tests/Tests/SkillsetsTests.cs]: > I have a concern about an apparently unrelated test change.

@Careyjmac and I have added extra context in the thread. Let me know what you think

**Comment by arv100kri** [comment]
> I have a concern about an apparently unrelated test change.

@Careyjmac and I have added extra context in the thread. Let me know what you think

**Review by bitsforthought — APPROVED** [review]

**Comment by danieljurek** [comment]
LGTM... preceding version bump looks good as well given this is a breaking change. Merging...

**Pull request merged**